src/transformers/models/speech_to_text/modeling_speech_to_text.py

```diff
@@ -0,0 +1,1348 @@
+
+    def _get_subsampled_output_lengths(self, input_lengths: torch.LongTensor):
+        """
+        Computes the output length of the convolutional layers
+        """
+
+        for i in range(self.config.num_conv_layers):
+            input_lengths = (input_lengths - 1) // 2 + 1
+
+        return input_lengths
+
+    def _get_subsampled_encoder_attn_mask(self, attention_mask):
+        # generate creates 3D attention mask, becuase of the shape of input_features
+        # convert it to 2D if thats the case
+        if len(attention_mask.shape) > 2:
+            attention_mask = attention_mask[:, :, -1]
+
+        subsampled_lengths = self._get_subsampled_output_lengths(attention_mask.sum(-1))
+        max_len = subsampled_lengths.max().item()
+        bsz = attention_mask.size()[0]
+        attention_mask = torch.zeros((bsz, max_len), dtype=attention_mask.dtype, device=attention_mask.device)
+
+        # these two operations makes sure that all values
+        # before the output lengths indices are attended to
+        attention_mask[(torch.arange(bsz, device=attention_mask.device), subsampled_lengths - 1)] = 1
+        attention_mask = attention_mask.flip([-1]).cumsum(-1).flip([-1]).long()
+        return attention_mask
+
+
+SPEECH_TO_TEXT_START_DOCSTRING = r"""
+    This model inherits from :class:`~transformers.PreTrainedModel`. Check the superclass documentation for the generic
+    methods the library implements for all its model (such as downloading or saving, resizing the input embeddings,
+    pruning heads etc.)
+
+    This model is also a PyTorch `torch.nn.Module <https://pytorch.org/docs/stable/nn.html#torch.nn.Module>`__
+    subclass. Use it as a regular PyTorch Module and refer to the PyTorch documentation for all matter related to
+    general usage and behavior.
+
+    Parameters:
+        config (:class:`~transformers.Speech2TextConfig`):
+            Model configuration class with all the parameters of the model. Initializing with a config file does not
+            load the weights associated with the model, only the configuration. Check out the
+            :meth:`~transformers.PreTrainedModel.from_pretrained` method to load the model weights.
+"""
+
+SPEECH_TO_TEXT_INPUTS_DOCSTRING = r"""
+    Args:
+        input_features (:obj:`torch.LongTensor` of shape :obj:`(batch_size, fbank_feature_length, num_mel_bins)`):
+            Float values of fbank features extracted from the raw speech waveform. Raw speech waveform can be obtained
+            by loading a `.flac` or `.wav` audio file into an array of type `List[float]` or a `numpy.ndarray`, *e.g.*
+            via the soundfile library (`pip install soundfile`). To prepare the array into `input_features`, the
+            :class:`Speech2TextTokenizer` should be used for extracting the fbank features, padding and conversion into
+            a tensor of type `torch.FloatTensor`. See :meth:`transformers.Speech2TextTokenizer.__call__`
```

Comment:
Please remember that:
- \`xxx\` does not mean code formatting in rst, you need to use \`\`xxx\`\` or :obj:\`xxx\`
- :class:\`Speech2TextTokenizer\` will not work, you have to write :class:\`~transformers.Speech2TextTokenizer\`